src/ctapipe/monitoring/interpolation.py

```diff
@@ -184,5 +209,133 @@ def add_table(self, tel_id, input_table):
         alt = input_table["altitude"].quantity.to_value(u.rad)
         mjd = input_table["time"].tai.mjd
         self._interpolators[tel_id] = {}
-        self._interpolators[tel_id]["az"] = interp1d(mjd, az, **self.interp_options)
-        self._interpolators[tel_id]["alt"] = interp1d(mjd, alt, **self.interp_options)
+        self._interpolators[tel_id]["az"] = interp1d(
+            mjd, az, kind="linear", **self.interp_options
+        )
+        self._interpolators[tel_id]["alt"] = interp1d(
+            mjd, alt, kind="linear", **self.interp_options
+        )
+
+
+class ChunkInterpolator(MonitoringInterpolator):
+    """
+    Simple interpolator for overlapping chunks of data.
+    """
+
+    def __init__(self, h5file: None | tables.File = None, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self._interpolators = {}
+        self.start_time = {}
+        self.end_time = {}
+        self.values = {}
+        self.columns = list(self.required_columns)  # these will be the data columns
+        self.columns.remove("start_time")
+        self.columns.remove("end_time")
+
+    def __call__(self, tel_id: int, time: Time) -> float | dict[str, float]:
+        """
+        Interpolate overlapping chunks of data for a given time, tel_id, and column(s).
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id.
+        time : astropy.time.Time
+            Time for which to interpolate the data.
+
+        Returns
+        -------
+        interpolated : float or dict
+            Interpolated data for the specified column(s).
+        """
+
+        self._check_interpolators(tel_id)
+
+        result = {}
+        mjd = time.to_value("mjd")
+        for column in self.columns:
+            result[column] = self._interpolators[tel_id](column, mjd)
+
+        if len(result) == 1:
+            return result[self.columns[0]]
+        return result
+
+    def add_table(self, tel_id: int, input_table: Table) -> None:
+        """
+        Add a table to this interpolator for specific columns.
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id.
+        input_table : astropy.table.Table
+            Table of values to be interpolated, expected columns
+            are ``start_time`` as ``validity start Time`` column,
+            ``end_time`` as ``validity end Time`` and the specified columns
+            for the data of the chunks.
+        """
+
+        self._check_tables(input_table)
+
+        input_table = input_table.copy()
+        input_table.sort("start_time")
+
+        if tel_id not in self._interpolators:
+            self._interpolators[tel_id] = {}
+            self.values[tel_id] = {}
+            self.start_time[tel_id] = input_table["start_time"].to_value("mjd")
+            self.end_time[tel_id] = input_table["end_time"].to_value("mjd")
+            self._interpolators[tel_id] = partial(self._interpolate_chunk, tel_id)
```

Comment:
Why are we still storing a lambda / partial here in `_interpolators`? Why not just call `self._interpolate_chunk` directly in `__call__`?